main.cpp

```diff
@@ -73,418 +73,233 @@ compile/link/run to make sure you don't have any errors
+        Numeric& apply(std::function<Numeric&(ValueType&)> callable)
         {
             if(callable)
             {
-                return callable(*heapDub); 
+                return callable(*heapNumber); 
             }
 
             return *this;
         }
-        DoubleType& apply( void(*funcPtr)(double&))
+
+        Numeric& apply( void(*funcPtr)(ValueType&) )
         {
             if(funcPtr)
             {
-                funcPtr(*heapDub);    
+                funcPtr(*heapNumber); 
             }
 
             return *this;
         }
 
     private:
-        double* heapDub = nullptr;
-        DoubleType& powInternal( double x );
+        std::unique_ptr<ValueType> heapNumber{ new ValueType() };
+        Numeric& powInternal( ValueType x )
+        {
+            if( heapNumber != nullptr )
+            {
+                *heapNumber = static_cast<ValueType>( std::pow( *heapNumber, x ) );
+            }
+            return *this;
+        }
 
 };
 
-// int UDT
-#include <iostream>
-struct IntType
+template<>
+struct Numeric<double>
 {
+    using ValueType = double;
     public:
-        IntType( int i ) : heapInt( new int(i) ) {}
-        ~IntType()
+        Numeric( ValueType n ) : heapNumber( new ValueType(n) ) {}
+        Numeric() : Numeric(0) {}
+
+        ~Numeric()
         {
-            delete heapInt;
-            heapInt = nullptr;
+            heapNumber = nullptr;
         }
 
-        operator int() const { return *heapInt; }
+        operator ValueType() const { return *heapNumber; }
 
-        IntType& operator+=( const int y )
+        Numeric& operator+=( const ValueType y )
         {
-            *heapInt += y;
+            *heapNumber += y;
             return *this;
         }
 
-        IntType& operator-=( const int y )
+        Numeric& operator-=( const ValueType y )
         {
-            *heapInt -= y;
+            *heapNumber -= y;
             return *this;
         }
 
-        IntType& operator*=( const int y )
+        Numeric& operator*=( const ValueType y )
         {
-            *heapInt *= y;
+            *heapNumber *= y;
             return *this;
         }
 
-        IntType& operator/=( const int y )
+        Numeric& operator/=( const ValueType y )
         {
-            if( y == 0 )
+            if( y == 0.0 ) 
             {
-                std::cout << "Can't divide by 0! Cancelling divide operation."  << std::endl;       
+                std::cout << "Can't divide by 0! Cancelling operation." << std::endl;
                 return *this;
             }
-    
-            *heapInt /= y;
+
+            *heapNumber /= y;
             return *this;
         }
 
-        IntType& pow( const FloatType& operand );
-        IntType& pow( const DoubleType& operand );
-        IntType& pow( const IntType& operand );
-        IntType& pow( int operand );
+        Numeric& pow( const Numeric operand )
+        {
+            return powInternal( operand );
+        }
```

Comment:
This takes an instance of `Numeric<double>` as the argument, not a primitive, and also copies that instance.  `Numeric` is not copyable.
It forces implicit conversion of `operand to `ValueType`, if you look at `powInternal`'s signature.
I invoked this implicit conversion in `main()` which triggered an error. 
- [ ] revise